test/integration/scheduler/workload_controller_test.go

```diff
@@ -69,6 +70,134 @@ var _ = ginkgo.Describe("Workload controller with scheduler", func() {
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).To(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check3, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check2, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor2, true)
+		})
+
+		ginkgo.It("the workload should have appropriate AdditionalChecks added", func() {
+			wl := testing.MakeWorkload("wl", ns.Name).
+				Queue("queue").
+				Request(resourceGPU, "3").
+				Obj()
+			wlKey := client.ObjectKeyFromObject(wl)
+
+			ginkgo.By("creating and reserving quota the workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).Should(gomega.Succeed())
+					g.Expect(workload.HasQuotaReservation(&updatedWl)).Should(gomega.BeTrue(), "should have quota reservation")
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+				gomega.Expect(workload.IsAdmitted(&updatedWl)).To(gomega.BeFalse())
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).To(gomega.Succeed())
+					checks := slices.Map(updatedWl.Status.AdmissionChecks, func(c *kueue.AdmissionCheckState) string { return c.Name })
+					g.Expect(checks).Should(gomega.ConsistOf("check1", "check2"))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("updating admission check strategies", func() {
+				createdQueue := kueue.ClusterQueue{}
+				queueKey := client.ObjectKeyFromObject(clusterQueue)
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, queueKey, &createdQueue)).To(gomega.Succeed())
+					createdQueue.Spec.AdmissionChecksStrategy.AdmissionChecks = []kueue.AdmissionCheckStrategyRule{
+						*testing.MakeAdmissionCheckStrategyRule("check1", kueue.ResourceFlavorReference(flavorOnDemand)).Obj(),
+						*testing.MakeAdmissionCheckStrategyRule("check2", kueue.ResourceFlavorReference(reservationFlavor)).Obj(),
+						*testing.MakeAdmissionCheckStrategyRule("check3").Obj()}
+					g.Expect(k8sClient.Update(ctx, &createdQueue)).To(gomega.Succeed())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
```

Comment:
No action from the test, so no worthy of a `By`

Reply:
Fixed